lib/metadata/event-debouncer.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+var utils = require('../utils');
+
+/**
+ * Debounce protocol events by acting on those events with a sliding delay.
+ * @param {Number} delay
+ * @ignore
+ * @constructor
+ */
+function EventDebouncer(delay) {
+  this._delay = delay;
+  this._queue = null;
+  this._timeout = null;
+}
+
+/**
+ * Adds a new event to the queue and moves the delay.
+ * @param {{ handler: Function, all: boolean|undefined, keyspace: String|undefined,
+ *  cqlObject: String|undefined, callback: Function|undefined }} event
+ */
+EventDebouncer.prototype.eventReceived = function (event) {
+  event.callback = event.callback || utils.noop;
+  this._queue = this._queue || { callbacks: [], keyspaces: {} };
+  //TODO: Warn when size is too large
```

Comment:
Another difference here is that a single event debouncer is shared between all kinds of events (schema refreshes and host refreshes).  I like this, keeps things much simpler, but now you can't have different interval configurations between the two kind of events, which isn't that important (I haven't heard of cases of people configuring those differently yet).

The java driver also debounces node refresh requests, which are triggered as result of a `STATUS_CHANGE` event.  This prevent a burst of UP/DOWN/UP from marking up down and up again, and instead just up.  Do you think there is any value in debouncing that?  It seems less important that node list / schema refresh.  Also the java driver refreshes the node info from peers on these events and the node driver does not (from what I can tell).